client/src/elm/App/View.elm

```diff
@@ -1,24 +1,25 @@
 module App.View exposing (..)
 
 import App.Model exposing (..)
-import App.Types exposing (Widget(..))
-import Pages.Homepage.View exposing (view)
-import Pages.Item.View exposing (view)
+import App.Types exposing (Page(..))
 import Html exposing (..)
 import Html.Attributes exposing (class)
+import Pages.Homepage.View
+import Pages.Item.View exposing (view)
 
 
 view : Model -> Html Msg
 view model =
-    case model.widget of
+    case model.activePage of
         HomePage ->
             div [ class "ui container" ]
-                [ Html.map MsgPagesHomepage <| Pages.Homepage.View.view model.baseUrl model.user model.pageHomepage
+                [ Pages.Homepage.View.view model.backendUrl model.user model.backend.items
                 ]
 
-        Item ->
+        Item itemId ->
             div [ class "ui container" ]
-                [ Html.map MsgPagesItem <| Pages.Item.View.view model.baseUrl model.user model.pageItem
+                -- @todo: Send model.backend instead of model.backend.items
+                [ Html.map MsgPagesItem <| Pages.Item.View.view model.backendUrl model.user model.backend.items itemId model.pagesItem
```

Comment:
It's going to be a trade-off with this approach, whether to send the whole `model.backend` or just some part of it, and then re-integrate what gets returned. 

- If the function needs several things from `model.backend`, it may make sense to send the whole thing.

- If the function can be limited to just some parts of `model.backend`, it could be nice to just send those parts ... that makes it nice to reason about what the function can see (and can modify).

Reply:
Yeah, I'm actually sending only the parts I need by having my function signature as `{ r | items : EntityDictList ... }`. It's just a todo to change that in that functions as-well :)

Reply:
(look for `partialBackendModel` in the code)

Reply:
Ah -- that is an interesting use of extensible records to define a kind of "partial" type ... I'll have to remember that one.